src/manager/frontend/src/client/util.ts

```diff
@@ -0,0 +1,16 @@
+export function getBaseURL(): string {
+  // If we are in development, use the local backend server
+  if (process.env.ENVIRONMENT === "dev") {
+    return "http://localhost:8000";
+  }
+  // Otherwise, in production, we can either:
+  // - Use the origin URL
+  // - Or use the NEXT_PUBLIC_API_URL environment variable if it is set (suitable for reverse proxy deployment)
+  if (typeof window !== "undefined") {
+    const w = window as any;
+    if (w.ENV && typeof w.ENV.NEXT_PUBLIC_API_URL === "string") {
```

Comment:
a little bit confusing to me that you call it `NEXT_PUBLIC_API_URL` as env variables starting with `NEXT_PUBLIC` have a special meaning in NextJS